src/urllib3/util/url.py

```diff
@@ -51,7 +51,7 @@
 ]
 
 UNRESERVED_PAT = r"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789._!\-~"
-IPV6_PAT = "(?:" + "|".join([x % _subs for x in _variations]) + ")"
+IPV6_PAT = "(?:" + "|".join(x % _subs for x in _variations) + ")"
```

Comment:
`"".join([...])` is faster than `"".join(...)`

Reply:
This is absolutely wild to me, but very true!

Had to see it for myself:

```ipython
In [6]: %timeit "(?:" + "|".join(x % _subs for x in _variations) + ")"
4.9 µs ± 26.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [7]: %timeit "(?:" + "|".join([x % _subs for x in _variations]) + ")"
4.66 µs ± 46.4 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```